src/test/resources/software/amazon/cloudformation/wrapper-override.json

```diff
@@ -1,12 +1,33 @@
 {
     "typeName": "Test::Resource::Type",
     "description": "Description",
+    "definitions": {
+        "subProperty": {
+            "type": "object",
+            "properties": {
```

Comment:
All right, I think the whole problem is that the schema doesn't define `"additionalProperties": false` for the nest property. In such way, the deserialization to POJO fails due to this mismatch. However, validation pass as `"additionalProperties": false` is missing and validation will skip this additional properties. 

The resource should fix the schema in such cases. But of course, this fix is still valuable.